### ISSUE TITLE
Problems with Ubuntu 22.04/DistUtils - Python 3.10+

### ISSUE THREAD

**Issue by ravenpi** [rcaloras/bashhub-client#109]
Distutils -- which, apparently, has been deprecated (https://docs.python.org/3/library/distutils.html) -- seems to have had some breaking changes, at least in Ubuntu 22.04:

ken@hal ~ $ rm -rf .bashhub/
ken@hal ~ $ curl -OL https://bashhub.com/setup && bash setup
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  8699  100  8699    0     0  36129      0 --:--:-- --:--:-- --:--:-- 36245
Using Python path /usr/bin/python3.9
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
100 10.7M  100 10.7M    0     0  4462k      0  0:00:02  0:00:02 --:--:-- 18.1M
Pulling down bashhub-client from 2.2.0 branch
Pulling down a few dependencies...(this may take a moment)
Traceback (most recent call last):
  File "/home/ken/.bashhub/bashhub-client-2.2.0/../env/bin/pip", line 5, in <module>
    from pip._internal.cli.main import main
  File "/home/ken/.bashhub/env/lib/python3.9/site-packages/pip/_internal/cli/main.py", line 9, in <module>
    from pip._internal.cli.autocompletion import autocomplete
  File "/home/ken/.bashhub/env/lib/python3.9/site-packages/pip/_internal/cli/autocompletion.py", line 10, in <module>
    from pip._internal.cli.main_parser import create_main_parser
  File "/home/ken/.bashhub/env/lib/python3.9/site-packages/pip/_internal/cli/main_parser.py", line 8, in <module>
    from pip._internal.cli import cmdoptions
  File "/home/ken/.bashhub/env/lib/python3.9/site-packages/pip/_internal/cli/cmdoptions.py", line 23, in <module>
    from pip._internal.cli.parser import ConfigOptionParser
  File "/home/ken/.bashhub/env/lib/python3.9/site-packages/pip/_internal/cli/parser.py", line 12, in <module>
    from pip._internal.configuration import Configuration, ConfigurationError
  File "/home/ken/.bashhub/env/lib/python3.9/site-packages/pip/_internal/configuration.py", line 26, in <module>
    from pip._internal.utils.logging import getLogger
  File "/home/ken/.bashhub/env/lib/python3.9/site-packages/pip/_internal/utils/logging.py", line 13, in <module>
    from pip._internal.utils.misc import ensure_dir
  File "/home/ken/.bashhub/env/lib/python3.9/site-packages/pip/_internal/utils/misc.py", line 40, in <module>
    from pip._internal.locations import get_major_minor_version, site_packages, user_site
  File "/home/ken/.bashhub/env/lib/python3.9/site-packages/pip/_internal/locations/__init__.py", line 14, in <module>
    from . import _distutils, _sysconfig
  File "/home/ken/.bashhub/env/lib/python3.9/site-packages/pip/_internal/locations/_distutils.py", line 9, in <module>
    from distutils.cmd import Command as DistutilsCommand
ModuleNotFoundError: No module named 'distutils.cmd'

Unsure where to go with this (I'm a Ruby guy, not Python)

**Comment by rcaloras** [comment]
@ravenpi thanks for reporting. Distutils may simply be not installed. Could you try installing it first and then rerunning setup:

```
sudo apt-get install python3.9-distutils

# Alternatively if specific 3.9 fails try
sudo apt-get install python3-distutils
```

There's one dependency which currently still relies on distutils, but will be removed in a future version of bashhub. Additional thread for reference: https://github.com/pypa/get-pip/issues/124#issuecomment-904380931

**Comment by ravenpi** [comment]
@rcaloras 
Thanks for the quick reply!  It is installed, but v. 3.10:

ken@hal ~ $ sudo apt-get install python3-distutils
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
python3-distutils is already the newest version (3.10.4-0ubuntu1).
The following packages were automatically installed and are no longer required:
  gir1.2-mutter-9 libexpat1-dev libllvm12 libmutter-9-0 libpython3-dev
  libpython3.10-dev python3-dev python3-wheel python3.10-dev
Use 'sudo apt autoremove' to remove them.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
ken@hal ~ $ python3 --version
Python 3.10.4
ken@hal ~ $ 
----------
Additionally, as I played with that system a fair bit to see if I could get it to work and am worried I might have polluted stuff, I tried an install on a pristine, set-up-minutes-ago Ubuntu 22.04 server:

ken@prairie ~ $ curl -OL https://bashhub.com/setup && bash setup
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  8699  100  8699    0     0  41956      0 --:--:-- --:--:-- --:--:-- 42024

 Sorry can't seem to find a version of python 3.5-3.10 or 2.7.9+ installed
ken@prairie ~ $ python3 --version
Python 3.10.4

I admit to some confusion, as it worked the charm on all my Ubuntu (and Ubuntu-variant) computers prior to 22.04.  (For the sheer hell of it, I also created a link, /usr/local/bin/python -> /usr/bin/python3, in case it just really needed an executable named "python", to no avail.)

**Comment by SoftCreatR** [comment]
I can confirm, that it doesn't work on Ubuntu 22.04 with Python 3.10.4 installed.

**Comment by rcaloras** [comment]
Hey guys, just put PR #111 on a dev branch which should fix this I believe and support Python 3.10. Mind giving a quick test? Would be much appreciated! @ravenpi @SoftCreatR 

You can run this fix on install by doing:

```bash
curl "https://bashhub.com/setup?version=2.3.0-dev" -o setup && bash setup 2.3.0-dev
```

**Comment by SoftCreatR** [comment]
Yes, that works. Thank you.

**Comment by rcaloras** [comment]
Cool, thanks for confirming the fix @SoftCreatR! I'll aim to merge and release early this week. @ravenpi please share feedback if you get a chance to test.

**Comment by ravenpi** [comment]
@rcaloras -- sorry; was away for the weekend.  Works great!!  Thanks so much!

**Comment by Baneeishaque** [comment]
> Hey guys, just put PR #111 on a dev branch which should fix this I believe and support Python 3.10. Mind giving a quick test? Would be much appreciated! @ravenpi @SoftCreatR
> 
> You can run this fix on install by doing:
> 
> ```shell
> curl "https://bashhub.com/setup?version=2.3.0-dev" -o setup && bash setup 2.3.0-dev
> ```

Works after `sudo apt-get install python3-distutils`

**Comment by casperdcl** [comment]
confirmed working for me (needed `python3-distutils`)
